kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -108,6 +108,34 @@ public Snapshot buildLatestSnapshot(TableClient tableClient, Path logPath, Path
         return getSnapshotAtInit(tableClient, logPath, dataPath);
     }
 
+    /**
+     * Construct the snapshot for the given table at the version provided.
+     * TODO add details about what errors it throws w.r.t. the version provided?
+     *
+     * @param tableClient Instance of {@link TableClient} to use.
+     * @param logPath     Where the Delta log files are located.
+     * @param dataPath    Where the Delta data files are located.
+     * @param version     The snapshot version to construct
+     * @return
+     * @throws TableNotFoundException
+     */
+    public Snapshot getSnapshotAt(
+            TableClient tableClient,
+            Path logPath,
```

Comment:
We should pass in `logPath` and `dataPath` into the constructor of SnapshotManager ... no need to keep passing them into every public API of SnapshotManager

Reply:
https://github.com/delta-io/delta/pull/2613/files separate PR